src/main/java/pl/edu/agh/firecell/engine/algorithm/TemperatureCalculation.java

```diff
@@ -0,0 +1,165 @@
+            MatterState upMatter = upCell.material().getMatterState();
+            Cell downCell = oldState.getCell(NeighbourUtils.down(cellIndex));
+            MatterState downMatter = downCell.material().getMatterState();
+
+            Cell eastCell = oldState.getCell(NeighbourUtils.east(cellIndex));
+            MatterState eastMatter = eastCell.material().getMatterState();
+            Cell westCell = oldState.getCell(NeighbourUtils.west(cellIndex));
+            MatterState westMatter = westCell.material().getMatterState();
+
+            if (northMatter == MatterState.SOLID && southMatter == MatterState.SOLID)
+                yTemp = computeConductivity(northCell, oldCell, southCell);
+            else if (northMatter == MatterState.SOLID)
+                yTemp = computeConductivity(northCell, oldCell, new Cell(oldCell));
+            else if (southMatter == MatterState.SOLID)
+                yTemp = computeConductivity(new Cell(oldCell), oldCell, southCell);
+
+            if (upMatter == MatterState.SOLID && downMatter == MatterState.SOLID)
+                zTemp = computeConductivity(upCell, oldCell, downCell);
+            else if (upMatter == MatterState.SOLID)
+                zTemp = computeConductivity(upCell, oldCell, new Cell(oldCell));
+            else if (downMatter == MatterState.SOLID)
+                zTemp = computeConductivity(new Cell(oldCell), oldCell, downCell);
+
+            if (eastMatter == MatterState.SOLID && westMatter == MatterState.SOLID)
+                xTemp = computeConductivity(eastCell, oldCell, westCell);
+            else if (eastMatter == MatterState.SOLID)
+                xTemp = computeConductivity(eastCell, oldCell, new Cell(oldCell));
+            else if (westMatter == MatterState.SOLID)
+                xTemp = computeConductivity(new Cell(oldCell), oldCell, westCell);
+
+        } catch (IndexOutOfBoundsException ignored) {
+            if (oldState.hasCell(NeighbourUtils.up(cellIndex))){
+                zTemp = computeConductivity(oldState.getCell(NeighbourUtils.up(cellIndex)), oldCell, new Cell(oldCell));
+            }
+        }
+        return yTemp + zTemp + xTemp;
+    }
+
+    private double computeConductivity(Cell former, Cell middle, Cell latter) {
+        return - CONDUCTIVITY_COEFFICIENT * (tempDiff(middle, former) + tempDiff(middle, latter));
+    }
+
+    private static double tempDiffAbs(Cell cellOne, Cell cellTwo) {
+        return Math.abs(tempDiff(cellOne, cellTwo));
+    }
+
+    private static double tempDiff(Cell cellOne, Cell cellTwo) {
+        return cellOne.temperature() - cellTwo.temperature();
+    }
+
+    private double computeConvection(State oldState, Cell oldCell, Vector3i cellIndex) {
+
+        final int HOTTER = 0;
+        final int COLDER = 1;
+
+        double fromDownToMe = 0;
+        double fromMeToUp = 0;
+
+        double fromMeToNeighbour = 0;
+        double toMeFromNeighbour = 0;
```

Comment:
These  variables are not needed, especially with this confusing names. Maybe just one, aggragting variable willl be enough?